lib/strings.js

```diff
@@ -26,44 +26,65 @@ const htmlEscape = (
   content.replace(HTML_ESCAPE_REGEXP, char => HTML_ESCAPE_CHARS[char])
 );
 
-const SUBST_REGEXP = /@([-.0-9a-zA-Z]+)@/g;
-
 const subst = (
   // Substitute variables
   tpl, // string, template body
   data, // hash, data structure to visualize
   dataPath, // string, current position in data structure
   escapeHtml // boolean, escape html special characters if true
   // Returns: string
-) => (
-  tpl.replace(SUBST_REGEXP, (s, key) => {
-    const pos = key.indexOf('.');
-    const name = pos === 0 ? dataPath + key : key;
-    let value = getByPath(data, name);
+) => {
+  let start = 0;
+  let end = tpl.indexOf('@');
+  if (end === -1) return tpl;
+
+  const defaultData = getByPath(data, dataPath);
+  let result = '';
+  while (end !== -1 && start < tpl.length) {
+    result = tpl.substring(start, end);
+    start = end + 1;
+    end = tpl.indexOf('@', start);
+    if (end === -1) {
+      start -= 1;
+      break;
+    }
+
+    let key = tpl.slice(start, end);
+    let d = data;
+    if (key.startsWith('.')) {
+      key = key.slice(1);
+      d = defaultData;
+    }
```

Comment:
```js
if (key.startsWith('.')) {
  key = key.slice(1);
  d = defaultData;
} else {
  key = tpl.slice(start, end);
  d = data;
}
```

Reply:
Slightly different but this indeed seems better.